test_runner/regress/test_wal_acceptor.py

```diff
@@ -853,11 +855,46 @@ def start_safekeeper(self, i):
             str(i),
             "--broker-endpoints",
             self.broker.client_url(),
-            "--daemonize",
         ]
+        log.info(f'Running command "{" ".join(cmd)}"')
 
-        log.info(f'Running command "{" ".join(args)}"')
-        return subprocess.run(args, check=True)
+        with open(safekeeper_dir / "safekeeper.log", "wb") as log_file:
```

Comment:
Can be extracted into separate function to avoid duplication

Reply:
Extracted, also rust code got under refactored in a similar way.